include/libcyphal/application/node/heartbeat.hpp

```diff
@@ -0,0 +1,198 @@
+
+    /// @brief Factory method to create a Heartbeat instance.
+    ///
+    /// @param presentation The presentation layer instance. In use to create 'Heartbeat' publisher.
+    /// @return The Heartbeat instance or a failure.
+    ///
+    static auto make(presentation::Presentation& presentation)
+        -> Expected<Heartbeat, presentation::Presentation::MakeFailure>
+    {
+        auto maybe_heartbeat_pub = presentation.makePublisher<Publisher::Message>();
+        if (auto* const failure = cetl::get_if<presentation::Presentation::MakeFailure>(&maybe_heartbeat_pub))
+        {
+            return std::move(*failure);
+        }
+
+        return Heartbeat{presentation, cetl::get<Publisher>(std::move(maybe_heartbeat_pub))};
+    }
+
+    Heartbeat(Heartbeat&& other) noexcept
+        : presentation_{other.presentation_}
+        , startup_time_{other.startup_time_}
+        , publisher_{other.publisher_}
+        , message_{other.message_}
+        , update_callback_fn_{std::move(other.update_callback_fn_)}
+        , next_exec_time_{other.next_exec_time_}
+
+    {
+        // We can't move `periodic_cb_` callback (b/c it captures its own `this` pointer),
+        // so we need to stop it in the moved-from object, and start in the new one.
+        other.stopPublishing();
+        startPublishing();
+    }
+
+    ~Heartbeat() = default;
+
+    Heartbeat(const Heartbeat&)                = delete;
+    Heartbeat& operator=(const Heartbeat&)     = delete;
+    Heartbeat& operator=(Heartbeat&&) noexcept = delete;
+
+    /// @brief Umbrella type for heartbeat update entities.
+    ///
+    struct UpdateCallback
+    {
+        /// @brief Defines standard arguments for heartbeat update callback.
+        ///
+        struct Arg
+        {
+            /// Holds current heartbeat message.
+            Message& message;
+
+            /// Holds the approximate time when the callback was called.
+            TimePoint approx_now;
+        };
+
+        /// @brief Defines signature of the heartbeat update callback function.
+        ///
+        /// Size of the function is arbitrary (4 pointers), but should be enough to simple lambdas.
+        ///
+        static constexpr std::size_t FunctionSize = sizeof(void*) * 4;
+        using Function                            = cetl::pmr::function<void(const Arg& arg), FunctionSize>;
+    };
+
+    /// @brief Sets the update callback for the heartbeat.
+    ///
+    /// @param update_callback_fn The update callback function, which will be called before publication of the next
+    ///                           heartbeat message. Allows to modify the `message` before it will be published.
+    ///                           `arg.message.uptime` field is automatically prepopulated to reflect duration
+    ///                           since the node startup, but application can modify it as well (if needed).
+    ///
+    void setUpdateCallback(UpdateCallback::Function&& update_callback_fn)
```

Comment:
We also need setters for `health`, `mode`, and `vendor_specific_status_code` in case one wants to update the fields asynchronously instead of using the callback.